### PR TITLE
composite-checkout: Disable plan length picker and delete button when form not ready

### PR DESCRIPTION
**Pull request by sirbrillig**
#### Changes proposed in this Pull Request

This disables the plan length picker and the item delete buttons in the review step of composite checkout when the form is not in a "ready" state (when it is loading, submitting, or validating).

#### Testing instructions

- Enter composite checkout from signup (otherwise the plan length picker will not display).
- Click through to the last step.
- Click to submit the form (it's ok if the submission fails). Verify that the plan length picker is disabled while the form is submitting.

### CODE REVIEW

**Comment by michaeldcain** [comment]
@fditrapani @sirbrillig: I added some disabled styling to the radio button and some logic (I think) to prevent switching to a disabled item. Should be ready for final review.

Screen Shot 2020-03-24 at 10 25 52 AM

File: packages/composite-checkout-wpcom/src/components/radio-button.js [hunk 1]
```diff
@@ -99,6 +99,8 @@ const RadioButtonWrapper = styled.div`
 	:hover svg {
 		filter: grayscale( 0 );
 	}
+
+	${handleWrapperDisabled};
```
Comment (sirbrillig) [inline packages/composite-checkout-wpcom/src/components/radio-button.js]: I was going to say that this seemed like a fragile pattern, but it [seems that it's also an official pattern](https://emotion.sh/docs/composition), so 👍. (Another option might be to create two styled components like `RadioButtonWrapper` and `RadioButtonWrapperDisabled`, the latter of which wraps the former, but I'm actually not sure what happens if you nest styled components that way.)

Comment (michaeldcain) [reply packages/composite-checkout-wpcom/src/components/radio-button.js]: Yeah, I used WordPress/components [styled components](https://github.com/WordPress/gutenberg/blob/master/packages/components/src/card/styles/card-styles.js#L30) as a pattern. I'd be happy to change this to improve it.

Comment (sirbrillig) [reply packages/composite-checkout-wpcom/src/components/radio-button.js]: Oh, nice. Even better. I'd say we should just keep it, since it's not unusual.

**Pull request merged**